### ISSUE TITLE
Remove development pip dependencies

### ISSUE THREAD

**Issue by albireox** [sdss/marvin#281]
Our `requirements.txt` includes many packages that are not needed for normal users. We have already started grouping some dependencies in `requirements_dev.txt` and `requirements_web.txt` but many more packages can be moved there. We should also create a `requirements_docs.txt` with Sphinx dependencies.

**Comment by havok2063** [comment]
I have done this in the mpl6 branch.  I think I've moved all non-essential dependencies into `requirements_web.txt`.  Installing `_web` will first install the `requirements.txt`.  The `_dev` list includes all the testing packages and first installs the `_web` list, so that developers will get everything.  I've modified the `.travis.yml` file to pip install the `_dev` list.  The main `setup.py` just installs the main `requirements.txt`.   Do we want to include new setup options to optionally pip install the web (or dev)?  This can be merged with the current flag for `noweb`.   I haven't created a `requirements_doc.txt` though.   

I tested this by importing every individual file in `core`, `tools`, `utils`.  Everything imports ok.  I have not yet tried to use any of the tools, nor have tried running the tests since that requires all the dependencies.
